lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/StakePools.hs

```diff
@@ -799,6 +823,79 @@ spec = do
         expectResponseCode HTTP.status403 rq
         expectErrorMessage (errMsg403WrongPool wrongPoolId) rq
 
+    it "STAKE_POOLS_JOIN/QUIT - Checking delegation expectations" $ \ctx -> do
+        (_, p1:p2:_) <- eventually $
+            unsafeRequest @[ApiStakePool] ctx Link.listStakePools Empty
+        w <- fixtureWallet ctx
+
+        request @ApiWallet ctx (Link.getWallet @'Shelley w) Default Empty >>= flip verify
+            [ expectFieldEqual #delegation notDelegatingNotJoining
+            ]
+
+        -- joining first pool
+        r1 <- joinStakePool ctx (p1 ^. #id) (w, fixturePassphrase)
+        expectResponseCode HTTP.status202 r1
+        eventually $ do
+            let ep = Link.listTransactions @'Shelley w
+            request @[ApiTransaction n] ctx ep Default Empty >>= flip verify
+                [ expectListItemFieldEqual 0 (#direction . #getApiT) Outgoing
+                , expectListItemFieldEqual 0 (#status . #getApiT) InLedger
+                ]
+        (currentEpochNo1, sp1) <- getSlotParams ctx
+        request @ApiWallet ctx (Link.getWallet @'Shelley w) Default Empty >>= flip verify
```

Comment:
This test whole test is very nice readable. Perhaps it would be more so if we created a helper
```haskell
let getWallet = request @ApiWallet ctx (Link.getWallet @'Shelley w) Default Empty
```
for re-use and brevity?

What do you think?

Reply:
yes, for sure it is good idea . But this is the challenge throughout all integration test. Maybe Piotr will tackle that in coming PRs cleaning this modules

Reply:
Fair enough.